cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -1246,3 +1280,16 @@ func (csr *ClusterStateRegistry) GetScaleUpFailures() map[string][]ScaleUpFailur
 	}
 	return result
 }
+
+// ShouldRegisterByNow returns true i.f.f. the given node should be registered by now, according to the provided time.
+func (csr *ClusterStateRegistry) ShouldRegisterByNow(nodeGroup cloudprovider.NodeGroup, un UnregisteredNode, currentTime time.Time) (bool, error) {
```

Comment:
nit: `ShouldBeProvisionedByNow`? I like this method much more than the existing approach of exposing the time limits and adding them in the call site. Could we wrap other cases in this and not expose the limits externally?

Reply:
Agreed, but I prefer to leave it out of this PR. Currently static_autoscaler uses MaxNodeProvisionTime in fixNodeGroupSize(), I'd prefer not to add too much scope to that. Added a TODO on MaxNodeProvisionTime and made maxNodeRegisterTime private.